torchvision/datasets/mnist.py

```diff
@@ -286,25 +288,175 @@ def download(self):
+            if not os.path.isfile(file_path):
+                download_url(url, root=self.raw_folder, filename=filename, md5=None)
+            files.append(file_path)
+
+        # process and save as torch files
+        print('Processing...')
+        data = read_sn3_pascalvincent_tensor(files[0])
+        assert(data.dtype == torch.uint8)
+        assert(data.ndimension() == 3)
+        targets = read_sn3_pascalvincent_tensor(files[1]).long()
+        assert(targets.ndimension() == 2)
+        if self.what == 'test10k':
+            data = data[0:10000,:,:].clone()
+            targets = targets[0:10000,:].clone()
+        if self.what == 'test50k':
+            data = data[10000:,:,:].clone()
+            targets = targets[10000:,:].clone()
+        with open(os.path.join(self.processed_folder, self.data_file), 'wb') as f:
+            torch.save((data, targets), f)
+
+    def __getitem__(self, index):
+        # redefined to handle the compat flag
+        img, target = self.data[index], self.targets[index]
+        img = Image.fromarray(img.numpy(), mode='L')
+        if self.transform is not None:
+            img = self.transform(img)
+        if self.compat:
+            target = int(target[0])
+        if self.target_transform is not None:
+            target = self.target_transform(target)
+        return img, target
+
+    def extra_repr(self):
+        return "Split: {}".format(self.what)
+
+
 def get_int(b):
     return int(codecs.encode(b, 'hex'), 16)
 
 
+def open_maybe_compressed_file(path):
+    """Return a file object that possibly decompresses 'path' on the fly.
+       Decompression occurs when argument `path` is a string and ends with '.gz' or '.xz'.
+    """
+    if type(path) != str: 
+        return path
+    if path.endswith('.gz'): 
+        return gzip.open(path, 'rb')
+    if path.endswith('.xz'):
+        return lzma.open(path, 'rb')
+    return open(path, 'rb')
+
+
+def read_sn3_pascalvincent_tensor(path):
+    """Read a SN3 file in "Pascal Vincent" format (Lush file 'libidx/idx-io.lsh').
+       Argument may be a filename, compressed filename, or file object.
+    """
+    ## typemap
+    if not hasattr(read_sn3_pascalvincent_tensor, 'typemap'):
+        read_sn3_pascalvincent_tensor.typemap = {
+            8:  ( torch.uint8, np.uint8, np.uint8 ),
+            9:  ( torch.int8, np.int8, np.int8 ),
+            11: ( torch.int16, np.dtype('>i2'), 'i2'),
+            12: ( torch.int32, np.dtype('>i4'), 'i4'),
+            13: ( torch.float32, np.dtype('>f4'), 'f4'),
+            14: ( torch.float64, np.dtype('>f8'), 'f8') }
+    ## read
+    with open_maybe_compressed_file(path) as f:
+        data = f.read()
+    ## parse
+    magic = get_int(data[0:4])
+    nd = magic % 256
+    ty = magic // 256
+    assert nd >= 1 and nd <= 3
+    assert ty >= 8 and ty <= 14
+    m = read_sn3_pascalvincent_tensor.typemap[magic // 256]
+    s = [ get_int(data[4*(i+1):4*(i+2)]) for i in range(nd) ]
```

Comment:
Can you fix lint here? We use spaces between operators

Reply:
Okay.